### DISCUSSION TITLE
Change default ui color in android app

### DISCUSSION THREAD

**Discussion by Korbi-code** [beeware/toga#1654] [Q&A] [answered]
Hello, 

I'm looking for a solution to change the default color for android. 

When the app is open the "formal_name" provided by the pyproject.toml file is written on a green background. 
Is it possible to change this color ? 

![pic](https://user-images.githubusercontent.com/53312348/200012099-48c7c352-ab4d-40d4-b52c-8d4ed651f071.jpg)

Thanks in advance

**Comment by freakboy3742** [comment, marked as answer]
This isn't currently configurable in a programmatic way; however, it is something we definitely want to make customizable. 

In the short term, if you want to modify this, you will need to modify the color values in the `app/src/main/res/values/colors.xml` file that is generated by Briefcase for your Android project. These values will be overwritten every time you regenerate the project; so if you want a specific set of values to be applied consistently, you may need to clone the template (https://github.com/beeware/briefcase-android-gradle-template), make the change in your copy, and modify `pyproject.toml` to point at that template.

If you look at the existing template, you'll also see that the background splash color is already templated and customizable; adding analogous "primary_color", "primary_color_dark" and "accent_color" options to the template, and exposing those options in Briefcase would be a welcome contribution.

**Reply by Korbi-code** [reply]
Hello, @freakboy3742 

Thanks a lot for the quick answer. 

Worked perfectly. 
Great project, love it. 



In the file "pyproject.toml" below [tool.briefcase.app.<ApplicationName>] I added the following line:
template = "Path/To/The/Cloned/BriefcaseAndroidGradleTemplateRepo"

**Reply by ikus060** [reply]
@freakboy3742 

Sorry to revive this question. But I'm wondering if this feature is now supported or still pending ?

**Reply by rmartin16** [reply]
Looks like this as implemented in https://github.com/beeware/briefcase-android-gradle-template/pull/77.

Documentation at https://briefcase.readthedocs.io/en/stable/reference/platforms/android/gradle.html#colors

**Reply by ikus060** [reply]
Thanks @rmartin16 for you help.

I've quickly go trough the documentation, but I fail to understand where I should define values for `base_theme`, `accent_color`, `primary_color`, etc. I'm guessing it need to be define in pyproject.yml but I'm not sure.

**Reply by rmartin16** [reply]
Yep; add them under the `android` section for the app in `pyproject.toml`.

**Reply by ikus060** [reply]
Thanks @rmartin16 It's working indeed. 

Could I make contribution to the documentation for this kind of little quirk ? I think it might help other people.

**Reply by rmartin16** [reply]
Absolutely; all contributions are welcome.

**Reply by freakboy3742** [reply]
@ikus060 Something to keep in mind if/when you draft that documentation is that we need to make a distinction between Briefcase and Toga configuration. Toga's docs shouldn't assume that the project will be deployed with Briefcase - it's definitely a reasonable assumption, but it shouldn't be an assumed default.
